docs/admin/templates/extending-templates/devcontainers.md

```diff
@@ -1,126 +1,345 @@
+  count            = data.coder_workspace.me.start_count
+  agent_id         = coder_agent.main.id
+  workspace_folder = "/home/coder/project/${module.git_clone[0].folder_name}"
+  depends_on       = [module.git_clone]
+}
+```
+
+</details>
+
+<details><summary>Kubernetes-based template (Sysbox runtime)</summary>
 
 ```terraform
 terraform {
   required_providers {
-    coder  = { source = "coder/coder" }
-    docker = { source = "kreuzwerker/docker" }
+    coder      = { source = "coder/coder" }
+    kubernetes = { source = "hashicorp/kubernetes" }
   }
 }
 
-provider "coder" {}
 data "coder_workspace" "me" {}
 data "coder_workspace_owner" "me" {}
 
-resource "coder_agent" "dev" {
-  arch                    = "amd64"
-  os                      = "linux"
+resource "coder_agent" "main" {
+  os   = "linux"
+  arch = "amd64"
+
   startup_script_behavior = "blocking"
-  startup_script          = "sudo service docker start"
-  shutdown_script         = "sudo service docker stop"
-  # ...
+  startup_script = "sudo service docker start"
 }
 
-module "devcontainers-cli" {
+module "devcontainers_cli" {
   count    = data.coder_workspace.me.start_count
   source   = "dev.registry.coder.com/modules/devcontainers-cli/coder"
-  agent_id = coder_agent.dev.id
+  agent_id = coder_agent.main.id
 }
 
-resource "coder_devcontainer" "my-repository" {
+module "git_clone" {
+  count    = data.coder_workspace.me.start_count
+  source   = "dev.registry.coder.com/modules/git-clone/coder"
+  agent_id = coder_agent.main.id
+  url      = "https://github.com/coder/coder.git"
+  base_dir     = "/home/coder/project"
+}
+
+resource "coder_devcontainer" "project" {
   count            = data.coder_workspace.me.start_count
-  agent_id         = coder_agent.dev.id
-  workspace_folder = "/home/coder/my-repository"
+  agent_id         = coder_agent.main.id
+  workspace_folder = "/home/coder/project/${module.git_clone[0].folder_name}"
+  depends_on       = [module.git_clone]
 }
 
-resource "docker_container" "workspace" {
+resource "kubernetes_pod" "workspace" {
   count = data.coder_workspace.me.start_count
-  image = "codercom/oss-dogfood:latest"
-  env = [
-    "CODER_AGENT_DEVCONTAINERS_ENABLE=true",
-    # ... Other environment variables.
-  ]
-  # ... Other container configuration.
+
+  metadata {
+    name       = "coder-${data.coder_workspace_owner.me.name}-${lower(data.coder_workspace.me.name)}"
+    namespace = "coder-workspaces"
+  }
+
+  spec {
+    container {
+      name  = "main"
+      image = "codercom/enterprise-base:ubuntu"
+
+      security_context { privileged = true }  # or use Sysbox / rootless
+      env { name = "CODER_AGENT_TOKEN" value = coder_agent.main.token }
+    }
+  }
 }
 ```
 
+</details>
+
+## Troubleshoot Common Issues
+
+### Disable dev containers integration
+
+To disable the dev containers integration in your workspace, add the `CODER_AGENT_DEVCONTAINERS_ENABLE` environment variable to your existing `coder_agent` block:
```

Comment:
This is incorrect, the env must be set _before_ the agent is started, i.e. on the Docker container. We could potentially implement this behavior but it's currently not supported (/cc @DanielleMaywood thoughts?)

Reply:
changed to

~~~md
To disable the dev containers integration in your workspace, set the `CODER_AGENT_DEVCONTAINERS_ENABLE= "false"` environment variable.
~~~